src/rtc/core.py

```diff
@@ -152,6 +153,304 @@ def save_as_cog(filename, scratch_dir = '.', logger = None,
+
+        return
+
+    rasters_to_check_dict = {'DEM': (dem_file_description, dem_file)}
+
+    tile_min_x_utm = geogrid.start_x
+    tile_max_y_utm = geogrid.start_y
+    tile_max_x_utm = geogrid.start_x + geogrid.spacing_x * geogrid.width
+    tile_min_y_utm = geogrid.start_y + geogrid.spacing_y * geogrid.length
+
+    tile_srs = osr.SpatialReference()
+    tile_srs.ImportFromEPSG(geogrid.epsg)
+
+    for file_type, (file_description, file_name) in \
+            rasters_to_check_dict.items():
+
+        # check if file was provided
+        if not file_name:
+            error_msg = f'ERROR {file_description} not provided'
+            logger.error(error_msg)
+            raise ValueError(error_msg)
+
+        # check if file exists
+        if not os.path.isfile(file_name):
+            error_msg = f'ERROR {file_description} not found: {file_name}'
+            logger.error(error_msg)
+            raise FileNotFoundError(error_msg)
+
+        # test if tile is fully covered by the ancillary input
+        # by checking all ancillary input vertices are located
+        # outside of the tile grid.
+        gdal_ds = gdal.Open(file_name, gdal.GA_ReadOnly)
+
+        file_geotransform = gdal_ds.GetGeoTransform()
+        file_projection = gdal_ds.GetProjection()
+        min_x, dx, _, max_y, _, dy = file_geotransform
+        file_width = gdal_ds.GetRasterBand(1).XSize
+        file_length = gdal_ds.GetRasterBand(1).YSize
+
+        del gdal_ds
+
+        max_x = min_x + file_width * dx
+        min_y = max_y + file_length * dy
+
+        file_srs = osr.SpatialReference()
+        file_srs.ImportFromProj4(file_projection)
+        tile_polygon, tile_min_y, tile_max_y, tile_min_x, tile_max_x = \
+            get_tile_srs_bbox(tile_min_y_utm, tile_max_y_utm,
+                              tile_min_x_utm, tile_max_x_utm,
+                              tile_srs, file_srs)
+
+        # Create input ancillary polygon
+        file_polygon = _get_ogr_polygon(min_x, max_y, max_x, min_y, file_srs)
+
+        coverage_logger_str = file_description+' coverage'
+        coverage_metadata_str = file_type+'_COVERAGE'
+
+        if tile_polygon.Within(file_polygon):
+            # print messages to the user
+            logger.info(f'    {coverage_logger_str}: Full')
+
+            # update DSWx-HLS product metadata
+            metadata_dict[coverage_metadata_str] = 'FULL'
+            continue
+
+        flag_error = False
+
+        # If needed, test for antimeridian ("dateline") crossing
+        if _antimeridian_crossing_requires_special_handling(
+                file_srs, min_x, tile_min_x, tile_max_x):
+
+            logger.info(f'The input HLS product crosses the antimeridian'
+                        ' (dateline). Verifying the'
+                        f' {file_description}: {file_name}')
+
+            # Left side of the antimeridian crossing: -180 -> +180
+            file_polygon_1 = _get_ogr_polygon(-180, 90, max_x, -90, file_srs)
```

Comment:
I think this line assumes that the ancillary file is in geographic coordinates. Can we always assume that the user is going to provide the DEM whose projection is sth. like EPSG 4326? What is going to happen when the input DEM is in UTM?

Reply:
Hi @seongsujeong , we first test if the ancillary file is in geographic coordinates within the function `_antimeridian_crossing_requires_special_handling()`, specifically in the test flag `flag_input_geographic_and_longitude_not_0_360`. Thank for checking!